### PR TITLE
Build and test with CUDA 12.5.1

### PR DESCRIPTION
**Pull request by KyleFromNVIDIA**
This PR updates the latest CUDA build/test version 12.2.2 to 12.5.1.

Contributes to https://github.com/rapidsai/build-planning/issues/73

### CODE REVIEW

**Comment by KyleFromNVIDIA** [comment]
Building this as a wheel requires a static `libnvidia-ml.a` (for now at least, until rapidsai/build-planning#35 is done), but CMake is not able to provide it. I've opened https://gitlab.kitware.com/cmake/cmake/-/issues/26131 to discuss this.

**Review by jameslamb — APPROVED** [review]
I've reviewed that offline discussion about handling `libnvidia-ml`, all makes sense to me. This looks good.

File: conda/recipes/libwholegraph/meta.yaml [hunk 1]
```diff
@@ -55,6 +55,7 @@ requirements:
     - cudatoolkit
     {% else %}
     - cuda-cudart-dev
+    - cuda-nvml-dev
```
Comment (bdice) [inline conda/recipes/libwholegraph/meta.yaml:58]: This change needed an entry in `dependencies.yaml` as well. See https://github.com/rapidsai/devcontainers/pull/366#issuecomment-2243859440. In devcontainers, we see:

```
2024-07-22T21:24:02.4790079Z CMake Error at CMakeLists.txt:204 (target_link_libraries):
2024-07-22T21:24:02.4793870Z   Target "wholegraph" links to:
2024-07-22T21:24:02.4798184Z 
2024-07-22T21:24:02.4802582Z     CUDA::nvml
2024-07-22T21:24:02.4804498Z 
2024-07-22T21:24:02.4804898Z   but the target was not found.  Possible reasons include:
2024-07-22T21:24:02.4805252Z 
2024-07-22T21:24:02.4805598Z     * There is a typo in the target name.
2024-07-22T21:24:02.4810296Z     * A find_package call is missing for an IMPORTED target.
2024-07-22T21:24:02.4811022Z     * An ALIAS target is missing.
```

Comment (bdice) [reply conda/recipes/libwholegraph/meta.yaml:58]: I filed https://github.com/rapidsai/wholegraph/pull/197 to fix.

Comment (jakirkham) [reply conda/recipes/libwholegraph/meta.yaml:58]: Good catch. Thanks Bradley! 🙏

**Pull request merged**